### PR TITLE
Move class ServiceBusJmsConnectionFactory into com/azure/spring/jms

### PR DESCRIPTION
**Pull request by moarychan**
# Description

Move the class `ServiceBusJmsConnectionFactory` into a new package `com.azure.spring.jms`, because it's a Spring JMS level class, instead of an autoconfiguration one.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/revapi/revapi.json [hunk 1]
```diff
@@ -357,6 +357,18 @@
...
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "interface jakarta\\.jms\\..*",
+          "justification": "They are external interfaces."
+        },
+        {
+          "regex": true,
+          "code": "java.class.externalClassExposedInAPI",
+          "new": "class jakarta\\.jms\\..*",
+          "justification": "They are external classes."
+        },
```
Comment (saragluna) [inline eng/code-quality-reports/src/main/resources/revapi/revapi.json:371]: Do we need this?

Comment (moarychan) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json:371]: Yes, I am trying to fix it. https://dev.azure.com/azure-sdk/public/_build/results?buildId=2055287&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=70ccc7c9-8d04-5255-456e-9568fe142bcd&l=13

Comment (moarychan) [reply eng/code-quality-reports/src/main/resources/revapi/revapi.json]: Removed.

**Comment by moarychan** [comment]
Closing it, changed in another PR https://github.com/Azure/azure-sdk-for-java/pull/32616

**Pull request closed without merging**